src/cata_tiles.cpp

```diff
@@ -263,9 +263,14 @@ cata::optional<tile_lookup_res>
 tileset::find_tile_type_by_season( const std::string &id, season_type season ) const
 {
     cata_assert( season < season_type::NUM_SEASONS );
-    const auto iter = tile_ids_by_season[season].find( id );
 
-    if( iter == tile_ids_by_season[season].end() ) {
+    const auto *tile_id = &tile_ids_by_season[season];
```

Comment:
You’d better use `.find()` on `tile_ids_by_season` and compare the iterator.

Reply:
I am not sure I understand what you mean, could you provide a code snippet to help me understand?

Reply:
Code refactored in https://github.com/CleverRaven/Cataclysm-DDA/pull/54826/commits/e831e7b8b6eded9e014a58d3074c3a5e5f1b2c28

Reply:
They are two different integrators and you seem to be misunderstanding my point.

Reply:
I apologize, I am quite new to C++ but I am trying to learn while helping develop the game I love.
This is why I asked for a snippet of code to help me understand exactly what you're referring to.

Reply:
Tiles ID by season is a data structure, and tiles IS by season [season] is another data structure. You want to check for whether season exists as a key in tiles IS by seasonal, not checking on tiles IS by season]season].

Reply:
I mean the check on whether season is a key in the outer data structure, not the inner data structure which current code is accessing. I tried typing code but give up, it's too difficult to type code on the phone.

Reply:
Does this not check if the given `id` exists as a key in `tile_ids_by_season`:
```c++
const auto iter = tile_ids_by_season->find( id );
if( iter == tile_ids_by_season->end() ) {
	debugmsg( "null `tile_ids_by_season` map entry for season: %d", season );
	return cata::nullopt;
}
``
I am retrieving an iterator to the key equal to `id` and then seeing if it there is a match.

Reply:
From what I can see in the code there doesn't seem to be any need to directly access map entry via index (season). I don't know why season is being passed as function argument here but it can be used for debug messaging if nothing else.

Reply:
If I understand correctly the intention of your pull request is to check first whether season is a key in the outer hash map, which is the exact missing check that causes the crash, and then the second step is to find ID in the inner hash map of that specific season. So you don't need to change the second step which is irrelevant to the bug you are addressing.

Reply:
After clarification on Discord I understand now what you meant.